### PR TITLE
New interface proposal

### PR DESCRIPTION
**Pull request by laliconfigcat**
### Describe the purpose of your pull request

This PR removes the ConfigCat-JS SDK dependency and uses only the configcat-common package. 
The ConfigCatProvider has the same parameters just like the create...Client functions in the configcat-common package. 
I added the tests for the ConfigCatProvider and for the HOC and Hooks variants too.
Coverage reporting is ready.

### Related issues (only if applicable)

Provide links to issues relating to this pull request

### Requirement checklist (only if applicable)

- [x] I have covered the applied changes with automated tests.
- [x] I have executed the full automated test set against my changes.
- [ ] I have validated my changes against all supported platform versions.
- [x] I have read and accepted the [contribution agreement](https://github.com/configcat/legal/blob/main/contribution-agreement.md).

### CODE REVIEW

File: src/ConfigCatProvider.tsx [hunk 1]
```diff
@@ -1,31 +1,86 @@
...
-      client: props.client,
-    });
+  private initializeConfigCatClient() {
+    let client: configcatcommon.IConfigCatClient;
+    switch (this.props.pollingMode) {
+      case PollingMode.LazyLoad:
+        client = configcatcommon.createClientWithLazyLoad(
+          this.props.sdkKey,
+          {
+            configFetcher: new HttpConfigFetcher(),
```
Comment (endret) [inline src/ConfigCatProvider.tsx:49]: It would be nice if you extracted the option into the const variable and reused it in all places.

**Review by endret — CHANGES_REQUESTED** [review]

File: src/setupTests.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
+// allows you to do things like:
```
[the remark this answers is not in the document]
Comment (laliconfigcat) [reply src/setupTests.ts]: Plans changed :D

File: src/ConfigCatProvider.tsx (hunk 1 above)
Comment (laliconfigcat) [reply src/ConfigCatProvider.tsx:44]: done

File: package.json [hunk 3]
```diff
@@ -3,21 +3,18 @@
...
+    "build:cjs": "tsc --module commonjs --outDir lib/cjs && gulp cjs",
+    "prepare": "npm run build",
+    "test": "react-scripts test"
   },
   "description": "ConfigCat is a configuration as a service that lets you manage your features and configurations without actually deploying new code.",
   "main": "lib/cjs/index.js",
   "types": "lib/esm/index.d.ts",
   "module": "lib/esm/index.js",
+  "files": [
+    "/lib"
```
Comment (endret) [inline package.json:16]: i guess you should add some pattern here like this:
```suggestion
    "/lib/**/*"
```

**Review by endret — CHANGES_REQUESTED** [review]

File: .npmignore [hunk 4]
```diff
@@ -8,4 +8,8 @@ package-lock.json
 tsconfig*.json
 .gitignore
 media/
-.github/
\ No newline at end of file
+.github/
+*.test.*
+update-samples.sh
+setupTests.*
+gulpfile.js
```
[the remark this answers is not in the document]
Comment (laliconfigcat) [reply .npmignore:15]: As i added the 
`files": [
    "/lib"`
part to the package.json this is not necessary. Actually the whole stuff could be fine-tuned. 
Checked the files with npm pack and everything seems alright.

File: package.json (hunk 3 above)
Comment (laliconfigcat) [reply package.json:16]: t works without this new pattern too. Checked with npm pack.

**Review by endret — APPROVED** [review]

**Review by sigewuzhere — APPROVED** [review]
Reviewed the readme :+1:

**Pull request merged**